test/arrayOracle.ts

```diff
@@ -362,13 +363,13 @@ describe('using Array as oracle', () => {
         let res: number[] = [];
         pipe(
           subject,
-          subscribe({
-            next: data => res.push(data),
-            error: () => assert.fail('should not call error'),
-            complete: () => {
-              completed = true;
+          subscribe<number>( // FIXME: fix subject's type then remove <number>
```

Comment:
Without this `<number>`, TypeScript would error on using `data` below because TS claimed it was `unknown`. We have to find a way of fixing subject.

Reply:
But again, your changes remove `| Callbag<A>` from the type of `pipe`, does it work if you don't remove that?

Reply:
hmm good point